pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -24,23 +24,24 @@ import (
-	parseStoredVersionErrMsg = "Cannot parse current Elasticsearch version. String format must be {major}.{minor}.{patch}[-{label}]"
-	parseVersionErrMsg       = "Cannot parse Elasticsearch version. String format must be {major}.{minor}.{patch}[-{label}]"
-	pvcImmutableErrMsg       = "volume claim templates can only have their storage requests increased, if the storage class allows volume expansion. Any other change is forbidden"
-	pvcNotMountedErrMsg      = "volume claim declared but volume not mounted in any container. Note that the Elasticsearch data volume should be named 'elasticsearch-data'"
-	unsupportedConfigErrMsg  = "Configuration setting is reserved for internal use. User-configured use is unsupported"
-	unsupportedUpgradeMsg    = "Unsupported version upgrade path. Check the Elasticsearch documentation for supported upgrade paths."
-	unsupportedVersionMsg    = "Unsupported version"
-	notAllowedNodesLabelMsg  = "Node label not in the exposed node labels list"
+	autoscalingVersionMsg              = "autoscaling is not available in this version of Elasticsearch"
+	cfgInvalidMsg                      = "Configuration invalid"
+	duplicateNodeSets                  = "NodeSet names must be unique"
+	invalidNamesErrMsg                 = "Elasticsearch configuration would generate resources with invalid names"
+	invalidSanIPErrMsg                 = "Invalid SAN IP address. Must be a valid IPv4 address"
+	masterRequiredMsg                  = "Elasticsearch needs to have at least one master node"
+	mixedRoleConfigMsg                 = "Detected a combination of node.roles and %s. Use only node.roles"
+	noDowngradesMsg                    = "Downgrades are not supported"
+	nodeRolesInOldVersionMsg           = "node.roles setting is not available in this version of Elasticsearch"
+	parseStoredVersionErrMsg           = "Cannot parse current Elasticsearch version. String format must be {major}.{minor}.{patch}[-{label}]"
+	parseVersionErrMsg                 = "Cannot parse Elasticsearch version. String format must be {major}.{minor}.{patch}[-{label}]"
+	pvcImmutableErrMsg                 = "volume claim templates can only have their storage requests increased, if the storage class allows volume expansion. Any other change is forbidden"
+	pvcNotMountedErrMsg                = "volume claim declared but volume not mounted in any container. Note that the Elasticsearch data volume should be named 'elasticsearch-data'"
+	unsupportedConfigErrMsg            = "Configuration setting is reserved for internal use. User-configured use is unsupported"
+	unsupportedUpgradeMsg              = "Unsupported version upgrade path. Check the Elasticsearch documentation for supported upgrade paths."
+	unsupportedVersionMsg              = "Unsupported version"
+	notAllowedNodesLabelMsg            = "Node label not in the exposed node labels list"
+	unsupportedClientAuthenticationMsg = "Mandatory client authentication is not supported"
```

Comment:
I think this can be confusing and it would be good to be more specific.
Something like `Configuration setting 'xpack.security.http.ssl.client_authentication: required' is unsupported`.

```go
func partiallyUnsupportedConfigErrMsg(setting, value string) string {
	return fmt.Sprintf("Configuration setting `%s: %s` is unsupported", setting, value)
}
```

Reply:
I forgot that we display `Spec.nodesets [0] .config.xpack.security.http.ssl.client_authentication` with the message. Without this information, it would have been confused.
It's good like that, don't change anything.